src/plugins/intel_cpu/src/nodes/dft.cpp

```diff
@@ -404,24 +794,182 @@ void DFT::naiveDFT(float* data, size_t dataLength) const {
     cpu_memcpy(data, outputBuffer.data(), dataLength * sizeof(float));
 }
 
-std::vector<std::pair<float, float>> DFT::generateTwiddles(size_t n_complex) const {
-    std::vector<std::pair<float, float>> twiddles(n_complex * n_complex);
+std::vector<float> DFT::DFTExecutor::generateTwiddlesDFT(size_t n_complex) const {
+    std::vector<float> twiddles(n_complex * n_complex * 2);
     parallel_for(n_complex, [&](const size_t k) {
         for (size_t n = 0; n < n_complex; ++n) {
-            float phase =  2.0f * PI * static_cast<float>(n * k) / static_cast<float>(n_complex);
+            float phase = 2.0f * PI * static_cast<float>(n * k) / static_cast<float>(n_complex);
             auto complexReal = std::cos(phase);
             auto complexImag = -std::sin(phase);
-            twiddles[k * n_complex + n] = std::make_pair(complexReal, complexImag);
+            twiddles[2 * (k * n_complex + n)] = complexReal;
+            twiddles[2 * (k * n_complex + n) + 1] = complexImag;
         }
     });
     return twiddles;
 }
 
+void DFT::DFTExecutor::generateTwiddlesFFT(size_t n_complex) {
+    size_t numBlocks = 1;
+
+    twiddlesFFT.reserve((n_complex - 1) * 2);
+    if (twiddlesFFT.size() == 0) {
+        twiddlesFFT.emplace_back(1.0f);   //  cos(0)
+        twiddlesFFT.emplace_back(-0.0f);  // -sin(0)
+    } else {
+        for (size_t i = numBlocks; i < twiddlesFFT.size() / 2; i += numBlocks) {
+            numBlocks *= 2;
+        }
+    }
+
+    for (size_t i = twiddlesFFT.size() / 2; i < n_complex - 1; i += numBlocks) {
+        numBlocks *= 2;
+
+        for (size_t blockNum = 0; blockNum < numBlocks; blockNum++) {
+            size_t copyIndex = twiddlesFFT.size() - blockNum - numBlocks;
+
+            twiddlesFFT.push_back(twiddlesFFT[copyIndex]);
+            twiddlesFFT.push_back(twiddlesFFT[copyIndex + 1]);
+
+            blockNum++;
+
+            float angle = PI * blockNum / numBlocks;
+            auto complexReal = std::cos(angle);
+            auto complexImag = -std::sin(angle);
+
+            twiddlesFFT.emplace_back(complexReal);
+            twiddlesFFT.emplace_back(complexImag);
+        }
+    }
+}
+
+DFT::DFTJitExecutor::DFTJitExecutor(const DFTAttrs& interpAttrs) : DFTExecutor(interpAttrs) {
+    jit_dft_config_params jdp = {};
+
+    jdp.inverse = interpAttrs.inverse;
+
+    if (mayiuse(cpu::x64::avx512_core)) {
```

Comment:
Kernels dftKernel and fftKernel are nod used simultaneously. So we need to generate code just for one. It will reduce first inference time and each inference time for dynamic shapes.